SwiftUIHomeworks/SwiftUIHomeworks/ViewModels/PlayerViewModel.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  PlayerViewModel.swift
+//  SwiftUIHomeworks
+//
+//  Created by Aleksandr Dorofeev on 19.01.2023.
+//
+
+import AVFoundation
+import SwiftUI
+
+/// View model for player screen
+final class PlayerViewModel: ObservableObject {
+  // MARK: - Private Constants
+  
+  private enum Constants {
+    static let formatString = "mp3"
+    static let zeroValue: Int = 0
+    static let oneValue = 1
+    static let zeroFloatValue: Float = 0
+    static let zeroTimeIntervalValue: TimeInterval = 0
+    static let scheduleTimeIntervalValue: TimeInterval = 0.2
+    static let amountHoursValue = 3600
+    static let amountMinutesValue = 60
+  }
+  
+  // MARK: - Public properties
+  
+  @Published var maxDuration = Float()
+  @Published var durationSong = Float()
+  
+  // MARK: - Private properties
+  
+  private let songs = Song.getSongs()
+  private var currentSongIndex = Constants.zeroValue
+  private var player: AVAudioPlayer?
+  
+  // MARK: - Public methods
+  
+  func play() {
+    currentSong(index: currentSongIndex)
+    player?.play()
+    countdown()
+  }
+  
+  func stop() {
+    player?.stop()
+    player?.currentTime = Constants.zeroTimeIntervalValue
+  }
+  
+  func setSongDuration(durationValue: Float) {
+    guard let time = TimeInterval(exactly: durationValue) else { return }
+    player?.currentTime = time
+    player?.play()
+    countdown()
+  }
+  
+  func nextSong() {
+    currentSongIndex += Constants.oneValue
+    durationSong = Constants.zeroFloatValue
+    currentSong(index: currentSongIndex)
+    player?.play()
+    countdown()
+  }
+  
+  func previousSong() {
+    currentSongIndex -= Constants.oneValue
+    durationSong = Constants.zeroFloatValue
+    currentSong(index: currentSongIndex)
+    player?.play()
+    countdown()
+  }
+  
+  func getCoverName() -> String {
+    songs[currentSongIndex].cover
+  }
+  
+  func getArtist() -> String {
+    songs[currentSongIndex].artist
+  }
+  
+  func getName() -> String {
+    songs[currentSongIndex].name
+  }
+  
+  func getCurrentTime() -> String {
+    return updateDuration(Int(durationSong))
+  }
+  
+  func getDuration() -> String {
+    return updateDuration(Int(maxDuration))
+  }
```

Comment:
```suggestion
  func getCurrentTime() -> String {
    updateDuration(Int(durationSong))
  }
  
  func getDuration() -> String {
    updateDuration(Int(maxDuration))
  }
```
 если компилятор не просит, не пишем